plugin/manager_linux.go

```diff
@@ -51,7 +51,32 @@ func (pm *Manager) enable(p *v2.Plugin, c *controller, force bool) error {
 }
 
 func (pm *Manager) restore(p *v2.Plugin) error {
-	return pm.containerdClient.Restore(p.GetID(), attachToLog(p.GetID()))
+	if err := pm.containerdClient.Restore(p.GetID(), attachToLog(p.GetID())); err != nil {
+		return err
+	}
+
+	if pm.liveRestore {
+		if pids, _ := pm.containerdClient.GetPidsForContainer(p.GetID()); len(pids) == 0 {
+			// plugin is not running, so follow normal startup procedure
+			return pm.enable(p, &controller{}, true)
+		}
+
+		c := &controller{restart: true, exitChan: make(chan bool)}
+		pm.mu.Lock()
+		pm.cMap[p] = c
+		pm.mu.Unlock()
+
+		client, err := plugins.NewClientWithTimeout("unix://"+filepath.Join(p.GetRuntimeSourcePath(), p.GetSocket()), nil, 0)
+		if err != nil {
+			c.restart = false
+			shutdownPlugin(p, c, pm.containerdClient)
+			return err
+		}
+
+		p.SetPClient(client)
```

Comment:
Can you refactor this into a function so that `enable` and `restore` can share code rather than have duplicate code?